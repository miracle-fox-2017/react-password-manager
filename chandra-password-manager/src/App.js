import React, { Component } from 'react';
import {BrowserRouter as Router, Route, Link} from 'react-router-dom'
import Navbar from './navbar'
import {addUser, getUser, hapususer, getUserSatuan, knop} from './actions/index'
import {connect} from 'react-redux'
import Formulir from './components/formbaru'
import Formulirlama from './components/formlama'

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
    }
  }

  componentWillMount () {
    this.props.getUser()
  }

  hapus (id) {
    // console.log('makan');
    this.props.hapususer(id)
  }

  edited (id) {
    this.props.getUserSatuan(id)
  }

  myFunction () {
    var input, filter, table, tr, td, i;
    input = document.getElementById("myInput");
    filter = input.value.toUpperCase();
    table = document.getElementById("myTable");
    tr = table.getElementsByTagName("tr");
    for (i = 0; i < tr.length; i++) {
      td = tr[i].getElementsByTagName("td")[1];
      if (td) {
        if (td.innerHTML.toUpperCase().indexOf(filter) > -1) {
          tr[i].style.display = "";
        } else {
          tr[i].style.display = "none";
        }
      }
    }
  }

  render() {
    return (
      <Router>
      <div className="App">
      <div className="container">

      <Route exact path="/" render={(props) => ( <Formulir/> )}/>
      <Route exact path="/:id" render={(props) => ( <Formulirlama/> )}/>

      <div>
      <div className="col-md-3">
      </div>
      <div className="col-md-6">
      <input className="form-control serach" type="text" id="myInput" onKeyUp={this.myFunction} placeholder="Search for username.."/>
      </div>
      </div>


      <div className="jarak">


        <table className="table table-hover" id="myTable">
        <thead>
          <tr>
          <th>Url</th>
          <th>Username</th>
          <th>Password</th>
          <th>Tanggal Daftar</th>
          <th>tanggal Edit</th>
          <th>Edit</th>
          <th>Delete</th>
          </tr>
        </thead>

        <tbody>
          {this.props.usersuccessget.map((data, index)=> {
            return(
              <tr>
              <td><a href={data.url}>{data.url}</a></td>
              <td>{data.username}</td>
              <td>{data.password}</td>
              <td>{data.createat}</td>
              <td>{data.editedat}</td>
              <td><Link to={'/' + data.id}><p onClick={()=> this.edited(data.id)}>Edit</p></Link></td>
              <td><a href="#"><p onClick={()=> this.hapus(data.id)}>Delete</p></a></td>
              </tr>
            )
          })}
        </tbody>
        </table>

        </div>

      </div>
      </div>
      </Router>
    );
  }
}

const mapState = state => {
  return {
    usersuccessget: state.usersuccessget.usersuccessget
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    getUser: () => dispatch(getUser()),
    hapususer: (id) => dispatch(hapususer(id)),
    getUserSatuan: (id) => dispatch(getUserSatuan(id))
  }
}

const app = connect(
  mapState,
  mapDispatchToProps
)(App)

export default app;
