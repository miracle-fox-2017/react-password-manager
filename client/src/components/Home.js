import React from 'react'
import { connect } from 'react-redux'
import { get_user_all, delete_user, getUserAPI, getUserByWord } from '../actions/index'
import firebase from 'firebase'
import { Route, Link } from 'react-router-dom'


class Home extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      url : '',
      username: '',
      password: '',
      createdAt: '',
      updatedAt: '',
      keyWord: ''
    }
    this.handleSearch = this.handleSearch.bind(this)
    this.submitSearch = this.submitSearch.bind(this)
  }

  handleSearch (event) {
    this.setState({
      keyWord : event.target.value
    })
  }

  submitSearch () {
    this.props.getUserByWord(this.state.keyWord)
  }

  delete (id) {
    this.props.delete_user(id);
  }

  componentWillMount () {
    this.props.getUserAPI()
  }

  render () {
    return (
        <div className="container-fluid">
           <div id="custom-search-input">
              <div className="input-group col-md-12">
                  <input type="text" className="  search-query form-control" placeholder="Search By USERNAME . . . . " onChange={this.handleSearch} value={this.state.keyWord}/>
                  <span className="input-group-btn">
                      <button className="btn btn-danger" type="button" onClick={this.submitSearch}>
                          <span className=" glyphicon glyphicon-search"></span>
                      </button>
                  </span>
              </div>
            </div>

            <h2>Basic Table</h2>
            <table className="table" border="2px">
              <thead>
                <tr>
                  <th>URL</th>
                  <th>USERNAME</th>
                  <th>PASSWORD</th>
                  <th>CREATEDAT</th>
                  <th>UPDATEAT</th>
                  <th colSpan="2">ACTION</th>
                </tr>
              </thead>
              <tbody>
              {this.props.user.map((data, index)=> {
                console.log('isi data', data);
                return(
                  <tr>
                  <td>{data.url}</td>
                  <td>{data.username}</td>
                  <td>{data.password}</td>
                  <td>{data.createdAt}</td>
                  <td>{data.updatedAt}</td>
                  <td><Link to={'/edit/' + data.id}><span className=" glyphicon glyphicon-pencil"></span></Link></td>
                  <td><a href="#"><p onClick={ ()=> this.delete(data.id)}><span className=" glyphicon glyphicon-trash"></span></p></a></td>
                  </tr>
                )
              })}
              </tbody>
            </table>
      </div>
    )
  }
}
const mapState = (state) => {
  console.log('INI DI STATE', state);
  return {
    user: state.Form.form
  }
}

const mapAction = (dispatch) => {
  return {
    getUserAPI: () => dispatch(getUserAPI()),
    delete_user: (id) => dispatch(delete_user(id)),
    getUserByWord: (url) => dispatch(getUserByWord(url))
  }
}

export default connect (mapState,mapAction)(Home)
