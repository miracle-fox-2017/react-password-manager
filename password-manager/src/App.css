.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color:#e65c00;
  height: 90px;
  padding: 20px;
  color: white;
}
form{
  margin-top: 20px;
}
.App-title {
  font-size: 1.5em;
}
.App-intro {
  font-size: large;
}
.row-centered {
  text-align:center;
}
.col-centered {
  display:inline-block;
  float:none;
  /* reset the text-align */
  text-align:left;
  /* inline-block space fix */
  margin-right:-4px;
  text-align: center;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
