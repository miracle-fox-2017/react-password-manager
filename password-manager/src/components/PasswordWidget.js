import React, { Component } from 'react';

class PasswordWidget extends Component {
  render() {

    return (
      <div style={{ margin: '20px 0px' }} className="text-center">
        <fieldset style={{ margin: "20px 0px" }} className="checkbox text-left">
          <p className="text-center">Password Strength :</p>
          <input type="checkbox" checked={this.props.upperCase} /> Password harus memiliki setidaknya satu karakter huruf besar (upper-case)<br />
          <input type="checkbox" checked={this.props.lowerCase} /> Password harus memiliki setidaknya satu karakter huruf kecil (lower-case)<br />
          <input type="checkbox" checked={this.props.specialCharacter} /> Password harus memiliki setidaknya satu karakter spesial (#$%@!%&..)<br />
          <input type="checkbox" checked={this.props.oneNumber} /> Password harus memiliki setidaknya satu angka<br />
          <input type="checkbox" checked={this.props.fiveChar} /> Password harus memiliki panjang (length) lebih dari 5 karakter<br />
        </fieldset>
      </div>
    )
  }
}

export default PasswordWidget
