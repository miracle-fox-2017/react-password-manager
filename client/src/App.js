import React, { Component } from 'react';
import { BrowserRouter as Router} from 'react-router-dom'
import Navbar from './components/Navbar'
import './App.css';
import { Provider } from 'react-redux'
import store from './store/index.js'


class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <Router>
          <div className="App">
            <Navbar/>
          </div>
        </Router>
      </Provider>
    )
  }
}

export default App;
