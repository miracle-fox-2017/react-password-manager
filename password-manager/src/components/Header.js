import React, { Component } from 'react';
import './../App.css';

class Header extends Component {
  render() {
    return (
      <div className="header text-center">
        <header className="App-header ">
          <h1 className="App-title">Password Manager</h1>
        </header>
      </div>
    );
  }
}

export default Header
