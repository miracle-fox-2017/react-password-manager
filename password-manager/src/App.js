import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom'
import { Provider } from 'react-redux'
import store from './store'
import './App.css';
import Home from './components/Home'
import DataListProfile from './components/DataListProfile'
import EditData from './components/EditData'

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <Router>
          <div>
            <Route exact path="/" component={Home} />
            <Route exact path="/list" component={DataListProfile} />
           <Route exact path="/list/:key" component={EditData} />
          </div>
        </Router>
      </Provider>
    );
  }
}

export default App;
