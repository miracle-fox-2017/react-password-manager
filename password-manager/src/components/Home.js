import React, { Component } from 'react';
import './../App.css';
import Header from './Header'
import { connect } from 'react-redux'
import { inputNewDataProfile } from '../actions/profileAction'
import ButtonListProfile from './ButtonListProfile'
import PasswordWidget from './PasswordWidget'

class Home extends Component {
  constructor() {
    super()

    this.state = {
      profile: {
        url: '',
        username: '',
        password: ''
      },
      upperCase: false,
      lowerCase: false,
      specialCharacter: false,
      oneNumber: false,
      fiveChar: false
    }
    this.handleInputChange = this.handleInputChange.bind(this)
    this.inputDataProfile = this.inputDataProfile.bind(this)
  }
  handleInputChange(event) {
    let profile = this.state.profile
    profile[event.target.name] = event.target.value
    this.setState({
      profile
    })
    let password = this.state.profile.password

    if (event.target.name === "password") {
      this.checkPassword(password)
    }
  }

  checkPassword(password) {
    if ((/[A-Z]/.test(password))) {
      this.setState({
        upperCase: true
      })
    } else {
      this.setState({
        upperCase: false
      })
    }
    if ((/[a-z]/.test(password))) {
      this.setState({
        lowerCase: true
      })
    } else {
      this.setState({
        lowerCase: false
      })
    }
    if ((/[$-/:-?{-~@#!"^_`[\]]/.test(password))) {
      this.setState({
        specialCharacter: true
      })
    } else {
      this.setState({
        specialCharacter: false
      })
    }
    if ((/[0-9]/.test(password))) {
      this.setState({
        oneNumber: true
      })
    } else {
      this.setState({
        oneNumber: false
      })
    }
    if (/.{6}$/.test(password)) {
      this.setState({
        fiveChar: true
      })
    } else {
      this.setState({
        fiveChar: false
      })
    }
  }
  inputDataProfile(event) {
    event.preventDefault()
    if (this.state.fiveChar && this.state.lowerCase && this.state.upperCase && this.state.oneNumber && this.state.specialCharacter) {
      this.props.inputDataProfile(this.state.profile)
      alert('Inserted 1 data...')
      this.setState({
        profile: {
          url: '',
          username: '',
          password: ''
        },
        fiveChar: false,
        lowerCase: false,
        upperCase: false,
        oneNumber: false,
        specialCharacter: false
      })
    } else {
      alert("Your password is not strength!")
    }
  }

  render() {

    return (
      <div className="App">
        <Header />
        <ButtonListProfile />
        <div className="container">
          <div className="row row-centered">
            <div className="col-md-6 col-centered">
              <form className="inputForm">
                <div className="form-group">
                  URL:
                    <input type="text" className="form-control" placeholder="Enter URL" name="url" value={this.state.profile.url} onChange={this.handleInputChange} />
                </div>
                <div className="form-group">
                  Username:
                    <input type="text" className="form-control" placeholder="Enter Username" name="username" value={this.state.profile.username} onChange={this.handleInputChange} />
                </div>
                <div className="form-group">
                  Password:
                    <input type="password" className="form-control" name="password" placeholder="Enter password" value={this.state.profile.password} onChange={this.handleInputChange} />
                </div>
              </form>
             <PasswordWidget upperCase={this.state.upperCase} lowerCase={this.state.lowerCase} oneNumber={this.state.oneNumber} specialCharacter={this.state.specialCharacter} fiveChar={this.state.fiveChar} onChange={this.handleInputChange}/>
              <button type="submit" className="btn btn-default" onClick={this.inputDataProfile}>Save</button>
            </div>
          </div>
        </div>
      </div >
    );
  }
}

const mapStateToProps = null
const mapDispatchToProps = (dispatch) => {
  return {
    inputDataProfile: (profile) => dispatch(inputNewDataProfile(profile))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)
