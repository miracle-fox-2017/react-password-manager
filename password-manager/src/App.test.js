import renderer from 'react-test-renderer'
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import store from './store'
import Home from './components/Home'
import Header from './components/Header'
import PasswordWidget from './components/PasswordWidget'
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders and match to snapshot', () => {
  const tree = renderer.create(<App />)
  expect(tree).toMatchSnapshot();
})


describe('Header component testing', () => {
  const wrapper = mount(<Header />)
  it('Should have div with class header', () => {
    expect(wrapper.find('div').hasClass('header text-center')).toBe(true)
  })
  it('Should have tag header with class app-header', ()=> {
    expect(wrapper.find('header').hasClass('App-header')).toBe(true)
  })
  it('Should have tag element h1 with title `Password Manager` ', ()=> {
    expect(wrapper.find('h1').text()).toBe('Password Manager')
  })
})

describe('Home component testing', () => {
  const wrapper = shallow(<Home/>, { context: { store } } ).dive()
  it('Should have form input with classs inputForm', () => {
    expect(wrapper.find('form').hasClass('inputForm')).toBe(true)
  })
  it('Should render 3', () => {
    expect(wrapper.find('input').length).toEqual(3)
  })
  it('Should have 3 forms input required', ()=> {
    expect(wrapper.find('input').at(0).prop('name')).toEqual('url')
    expect(wrapper.find('input').at(1).prop('name')).toEqual('username')
    expect(wrapper.find('input').at(2).prop('name')).toEqual('password')
  })
  it('Should return value', () => {
    expect(wrapper.find('input').at(0).value = "facebook.com").toEqual('facebook.com')
    expect(wrapper.find('input').at(1).value = "amelia").toEqual('amelia')
    expect(wrapper.find('input').at(2).value = "4meL!a").toEqual('4meL!a')
  })
})

describe('Password widget validation', ()=> {
  const wrapperHomeInput = shallow(<Home/>, { context: { store } } ).dive().find('input').at(2)
  const initialState = {
    upperCase: true,//false
    lowerCase: true,
    oneNumber: true,
    specialCharacter: true,
    fiveChar: true
  }
  // console.log(wrapperHomeInput)
  const wrapper = shallow(<PasswordWidget upperCase={initialState.upperCase} lowerCase={initialState.lowerCase} oneNumber={initialState.oneNumber} specialCharacter
  ={initialState.specialCharacter} fiveChar={initialState.fiveChar}/>)

  wrapper.setProps({password:'4meliaR@' })
  wrapper.simulate('change', {props: {password: '4meliaR@'}})
  const wrapperInput = wrapper.find('input')
  
  it('Should have at least 1 uppercase word', ()=> {
    expect(wrapperInput.at(0).props().checked).toEqual(true)
  })
  it('Should have at least 1 lowercase word', ()=> {
    expect(wrapperInput.at(1).props().checked).toEqual(true)
  })
  it('Should have at least 1 special character', ()=> {
    expect(wrapperInput.at(2).props().checked).toEqual(true)
  })
  it('Should have more than 1 number', ()=> {
    expect(wrapperInput.at(3).props().checked).toEqual(true)
  })
  it('Should have more than 5 character', ()=> {
    expect(wrapperInput.at(4).props().checked).toEqual(true)
  })
})

