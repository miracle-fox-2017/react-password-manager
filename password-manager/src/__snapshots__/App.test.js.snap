// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and match to snapshot 1`] = `
<div>
  <div
    className="App"
  >
    <div
      className="header text-center"
    >
      <header
        className="App-header "
      >
        <h1
          className="App-title"
        >
          Password Manager
        </h1>
      </header>
    </div>
    <div
      className="text-center"
      style={
        Object {
          "margin": "20px 0px",
        }
      }
    >
      <a
        href="/list"
        onClick={[Function]}
      >
        <button
          className="btn btn-primary"
          type="button"
        >
          Data List
        </button>
      </a>
    </div>
    <div
      className="container"
    >
      <div
        className="row row-centered"
      >
        <div
          className="col-md-6 col-centered"
        >
          <form
            className="inputForm"
          >
            <div
              className="form-group"
            >
              URL:
              <input
                className="form-control"
                name="url"
                onChange={[Function]}
                placeholder="Enter URL"
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              Username:
              <input
                className="form-control"
                name="username"
                onChange={[Function]}
                placeholder="Enter Username"
                type="text"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              Password:
              <input
                className="form-control"
                name="password"
                onChange={[Function]}
                placeholder="Enter password"
                type="password"
                value=""
              />
            </div>
          </form>
          <div
            className="text-center"
            style={
              Object {
                "margin": "20px 0px",
              }
            }
          >
            <fieldset
              className="checkbox text-left"
              style={
                Object {
                  "margin": "20px 0px",
                }
              }
            >
              <p
                className="text-center"
              >
                Password Strength :
              </p>
              <input
                checked={false}
                type="checkbox"
              />
               Password harus memiliki setidaknya satu karakter huruf besar (upper-case)
              <br />
              <input
                checked={false}
                type="checkbox"
              />
               Password harus memiliki setidaknya satu karakter huruf kecil (lower-case)
              <br />
              <input
                checked={false}
                type="checkbox"
              />
               Password harus memiliki setidaknya satu karakter spesial (#$%@!%&..)
              <br />
              <input
                checked={false}
                type="checkbox"
              />
               Password harus memiliki setidaknya satu angka
              <br />
              <input
                checked={false}
                type="checkbox"
              />
               Password harus memiliki panjang (length) lebih dari 5 karakter
              <br />
            </fieldset>
          </div>
          <button
            className="btn btn-default"
            onClick={[Function]}
            type="submit"
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
